### ISSUE TITLE
Type mismatch: inferred type is Context? but Context was expected

### ISSUE THREAD

**Issue by nilotpalkapri** [csdcorp/speech_to_text#259]
/snap/flutter/common/flutter/.pub-cache/hosted/pub.dartlang.org/speech_to_text-5.1.0/android/src/main/kotlin/com/csdcorp/speech_to_text/SpeechToTextPlugin.kt: (439, 58): Type mismatch: inferred type is Context? but Context was expected

FAILURE: Build failed with an exception.

**5.0.0 was working fine
**$ flutter doctor --verbose
[✓] Flutter (Channel beta, 2.5.0-5.3.pre, on Linux Mint 20.2 5.4.0-84-generic, locale en_IN)
    • Flutter version 2.5.0-5.3.pre at /home/nilotpalkapri/snap/flutter/common/flutter
    • Upstream repository https://github.com/flutter/flutter.git
    • Framework revision a7fb06d6fa (11 days ago), 2021-09-02 09:07:14 -0700
    • Engine revision fcde2c8b2f
    • Dart version 2.14.0 (build 2.14.0-377.8.beta)

[✓] Android toolchain - develop for Android devices (Android SDK version 31.0.0)
    • Android SDK at /home/nilotpalkapri/Android/Sdk
    • Platform android-31, build-tools 31.0.0
    • Java binary at: /snap/android-studio/114/android-studio/jre/bin/java
    • Java version OpenJDK Runtime Environment (build 11.0.10+0-b96-7249189)
    • All Android licenses accepted.

[✓] Chrome - develop for the web
    • Chrome at google-chrome

[✓] Linux toolchain - develop for Linux desktop
    • clang version 6.0.0-1ubuntu2 (tags/RELEASE_600/final)
    • cmake version 3.10.2
    • ninja version 1.8.2
    • pkg-config version 0.29.1

[✓] Android Studio (version 4.2)
    • Android Studio at /snap/android-studio/113/android-studio
    • Flutter plugin version 58.0.1
    • Dart plugin version 202.8531
    • Java version OpenJDK Runtime Environment (build 11.0.8+0-b944-P17168821)

[✓] Android Studio (version 2020.3)
    • Android Studio at /snap/android-studio/114/android-studio
    • Flutter plugin version 60.1.2
    • Dart plugin version 203.8292
    • Java version OpenJDK Runtime Environment (build 11.0.10+0-b96-7249189)

[✓] VS Code
    • VS Code at /snap/code/current
    • Flutter extension version 3.14.1

[✓] Connected device (3 available)
    • Redmi Note 5 Pro (mobile) • 46f54288 • android-arm64  • Android 9 (API 28)
    • Linux (desktop)           • linux    • linux-x64      • Linux Mint 20.2 5.4.0-84-generic
    • Chrome (web)              • chrome   • web-javascript • Google Chrome 93.0.4577.63

• No issues found!

**Comment by sowens-csd** [comment]
Thanks for the quick reporting on that. I'm having a look now.

**Comment by sowens-csd** [comment]
Interesting, it's working for me on Android. I see you're on a later Flutter version, so I tried updating to 2.5.0 and that also works on Android. Could also be something with an updated version of Kotlin in Android Studio. I've been meaning to do that for a while so I'll try that next.

**Comment by sowens-csd** [comment]
I can't reproduce this. I did a full upgrade of Android Studio and the various plugins. No problem building and running from either Flutter or Android Studio. Did you try a `flutter clean` and a rebuild? The line that's reference, and the code near it, hasn't changed from 5.0.0 to 5.1.0.

**Comment by nilotpalkapri** [comment]
Sorry for a bit late response, as I observed, the issue is coming when target sdk version is 31.
It is not related to this specific version. Targeting 30 is working fine.

**Comment by sowens-csd** [comment]
Thanks for following up. So all you did is set the `targetSdkVersion` in `build.gradle`?  I just tried that and it still compiles and deploys. Can you let me know what version of Kotlin and the Kotlin plugin you are using? Any other changes you can think of? 

```
    defaultConfig {
        applicationId "com.csdcorp.app.stt_5"
        minSdkVersion 21
        targetSdkVersion 31
        versionCode flutterVersionCode.toInteger()
        versionName flutterVersionName
    }
```

**Comment by nilotpalkapri** [comment]
During previous exception kotlin version was 1.5.20, upgraded it to 1.5.30, still throwing this error when target sdk is 31. 
Next will try creating a new project.

**Comment by sowens-csd** [comment]
I still can't reproduce, but...I'm sure there's a real issue here. So to try to speed up the process I've put in a potential fix for it, even though either the new or previous versions compile for me. Could you try to build against the current repo version and see if it resolves the issue? If so I'll put out a 5.2.0 build with this change.

**Comment by nilotpalkapri** [comment]
Yes its working. You can release this version.

**Comment by nilotpalkapri** [comment]
But please check one thing if it is necessary to put return in case of localContext==null

**Comment by sowens-csd** [comment]
Thanks for your help on this. It's now published as 5.1.1.
